main_test.go

```diff
@@ -355,6 +354,22 @@ func Test_Main_SampleXML(t *testing.T) {
 	// last span is server type
 	aTestCase = spans[expectedSpansCount-1]
 	assert.Equal(t, "SPAN_KIND_SERVER", aTestCase.Kind)
+
+	// metrics
+	resourceMetrics := testReport.resourceMetrics.Metrics[0]
+	metrics := resourceMetrics.InstrumentationLibraryMetrics[0].Metrics
+
+	assert.Equal(t, "tests.suite.error", metrics[0].Name)
```

Comment:
Two thoughts on semi-magic constants:

* these strings (eg "tests.suite.error") look like they're defined as consts in `semconv.go`. Should we use those consts instead?
* does it make sense to create a map from those strings to the expected indexes?

With those changes this line could be updated to something like:
`assert.Equal(t, ErrorTestsCount, metrics[MetricsIndex[ErrorTestsCount]].Name)`

that would also let this whole block of asserts get put into a loop.